tests/zerocashTest.cpp

```diff
@@ -72,6 +73,153 @@ int AddressTest() {
     return result;
 }
 
+int SaveAndLoadKeysFromFiles(int tree_depth) {
+    cout << "\nSaveAndLoadKeysFromFiles TEST\n" << endl;
+
+    cout << "Creating Params...\n" << endl;
+
+    libzerocash::timer_start("Param Generation");
+    auto keypair = libzerocash::ZerocashParams::GenerateNewKeyPair(tree_depth);
+    libzerocash::ZerocashParams p(
+        tree_depth,
+        &keypair
+    );
+    libzerocash::timer_stop("Param Generation");
+    print_mem("after param generation");
+
+    cout << "Successfully created Params.\n" << endl;
+
+    std::string vk_path = "./zerocashTest-verification-key";
+    std::string pk_path = "./zerocashTest-proving-key";
+
+    libzerocash::timer_start("Saving Proving Key");
+
+    libzerocash::ZerocashParams::SaveProvingKeyToFile(
+        &p.getProvingKey(),
+        pk_path
+    );
+
+    libzerocash::timer_stop("Saving Proving Key");
+
+    libzerocash::timer_start("Saving Verification Key");
+
+    libzerocash::ZerocashParams::SaveVerificationKeyToFile(
+        &p.getVerificationKey(),
+        vk_path
+    );
+
+    libzerocash::timer_stop("Saving Verification Key");
+
+    libzerocash::timer_start("Loading Proving Key");
+    auto pk_loaded = libzerocash::ZerocashParams::LoadProvingKeyFromFile(pk_path, tree_depth);
+    libzerocash::timer_stop("Loading Proving Key");
+
+    libzerocash::timer_start("Loading Verification Key");
+    auto vk_loaded = libzerocash::ZerocashParams::LoadVerificationKeyFromFile(vk_path, tree_depth);
+    libzerocash::timer_stop("Loading Verification Key");
+
+    cout << "Comparing Proving and Verification key.\n" << endl;
+
+    if ( !( p.getProvingKey() == pk_loaded && p.getVerificationKey() == vk_loaded) ) {
+        return false;
+    }
```

Comment:
[does not block] This looks great. I'm tempted to end a test on this line, then define a second test using the logic below. The second could call the first, or they could call a new common function.

Either both would fail or only the second longer test would fail. This would give us finer grained resolution on failures. Maybe not worth the extra running time, so maybe an alternative is to just rely on logging to determine which phase the single monolithic test is in.